cmdstanpy/stanfit.py

```diff
@@ -1457,3 +1337,121 @@ def save_csvfiles(self, dir: str = None) -> None:
         :param dir: directory path
         """
         self.runset.save_csvfiles(dir)
+
+
+def from_csv(
+    dir: str = None, method: str = None
+) -> Union[CmdStanMCMC, CmdStanMLE, CmdStanVB]:
```

Comment:
This is a pretty clear place where [typing literals and overloading](https://mypy.readthedocs.io/en/stable/literal_types.html) would be very helpful. Any users of mypy will appreciate if we annotate this in such a way that the return type isn't a union in the unambiguous cases

Reply:
if I'm understanding correctly, you're suggesting the following?

```
SampleMethod = Literal["sample"]
OptimizeMethod = Literal["optimize"]
@overload
def from_csv(dir: str, method: SampleMethod) -> CmdStanMCMC:
@overload
def from_csv(dir: str, method: OptimizeMethod) -> CmdStanMLE:

def from_csv(
    dir: str = None, method: str = None
) -> Union[CmdStanMCMC, CmdStanMLE, CmdStanVB]:
```

Reply:
This is my only remaining comment.

Here is what I believe a useful set of type signatures would be:

```python
# add this to the existing imports
from typing import Literal, overload

@overload
def from_csv(dir: str, method: Literal["sample"]) -> CmdStanMCMC: ...
@overload
def from_csv(dir: str, method: Literal["optimize"]) -> CmdStanMLE: ...
@overload
def from_csv(dir: str, method: Literal["variational"]) -> CmdStanVB: ...
@overload
def from_csv(dir: str, method: str) -> Union[CmdStanMCMC, CmdStanMLE, CmdStanVB]: ...

# this one is as existing and has actual implementation
def from_csv(
    dir: str = None, method: str = None
) -> Union[CmdStanMCMC, CmdStanMLE, CmdStanVB]:
```